### PR TITLE
aprox13 7/∞

### PR DESCRIPTION
**Pull request by maxpkatz**
This change splits the heavy ion reactions C12 + O16 and O16 + O16 into their respective endpoints. The former can end up with both Si28 and Mg24 + He4 with equal probability (same as iso7), while the latter can end in Si28 + He4 or S32, with an intermediate state of P31 + proton.

### CODE REVIEW

File: networks/aprox13/actual_rhs.H [hunk 1]
```diff
@@ -108,14 +108,38 @@ void aprox13tab(const Real btemp, const Real bden, Array1D<rate_t, 1, Rates::Num
...
+        case (O16_O16_to_Si28_He4_reverse):
+            dtab = bden; // rate is zero in this net
+            break;
+
+        case (O16_O16_to_S32_forward):
             dtab = bden;
             break;
 
+        case (O16_O16_to_S32_reverse):
+            dtab = bden; // rate is zero in this net
```
Comment (zingale) [inline networks/aprox13/actual_rhs.H:140]: isn't this 1?  Since it is just proportional to X(Si32)?

File: networks/aprox13/actual_rhs.H [hunk 2]
```diff
@@ -108,14 +108,38 @@ void aprox13tab(const Real btemp, const Real bden, Array1D<rate_t, 1, Rates::Num
...
+            break;
+
+        case (C12_O16_to_Si28_forward):
+            dtab = bden;
+            break;
+
+        case (C12_O16_to_Si28_reverse):
+            dtab = bden; // rate is zero in this net
+            break;
```
Comment (zingale) [inline networks/aprox13/actual_rhs.H:125]: shouldn't this be 1?

Comment (maxpkatz) [reply networks/aprox13/actual_rhs.H:125]: Fixed

File: networks/aprox13/actual_rhs.H [hunk 3]
```diff
@@ -1245,23 +1296,22 @@ void dfdy_isotopes_aprox13(Array1D<Real, 1, NumSpec> const& y,
 
     {
         // (he4)/d(o16)
-        Array1D<Real, 1, 5> b {
-            0.5e0_rt * y(C12) * rr(1).rates(C12_O16_forward),
-            1.12e0_rt * 0.5e0_rt*y(O16) * rr(1).rates(O16_O16_forward),
-            0.68e0_rt * rr(1).rates(irs1) * 0.5e0_rt*y(O16) * rr(1).rates(O16_O16_forward),
+        Array1D<Real, 1, 4> b {
+            y(C12) * rr(1).rates(C12_O16_to_Mg24_He4_forward),
+            y(O16) * rr(1).rates(O16_O16_to_Si28_He4_forward),
```
Comment (zingale) [inline networks/aprox13/actual_rhs.H:1301]: shouldn't there be a 2* here from differentiating y(O16)**2?

File: networks/aprox13/actual_rhs.H [hunk 4]
```diff
@@ -1485,15 +1538,14 @@ void dfdy_isotopes_aprox13(Array1D<Real, 1, NumSpec> const& y,
         jac(Mg24,Mg24) = esum3(b);
 
         // d(si28)/d(o16)
-        b(1) = 0.5e0_rt * y(C12) * rr(1).rates(C12_O16_forward);
-        b(2) = 1.12e0_rt * 0.5e0_rt*y(O16) * rr(1).rates(O16_O16_forward);
-        b(3) = 0.68e0_rt * 0.5e0_rt*y(O16) * rr(1).rates(irs1) * rr(1).rates(O16_O16_forward);
+        b(1) = y(C12) * rr(1).rates(C12_O16_to_Si28_forward);
+        b(2) = y(O16) * rr(1).rates(O16_O16_to_Si28_He4_forward);
```
Comment (zingale) [inline networks/aprox13/actual_rhs.H:1542]: also 2* here?

File: networks/aprox13/actual_rhs.H (hunk 3 above)
Comment (maxpkatz) [reply networks/aprox13/actual_rhs.H:1301]: It is implicitly there, notice the 0.5 in the middle of the term that is now gone

Comment (maxpkatz) [reply networks/aprox13/actual_rhs.H]: OK I agree I was missing a factor of 2 and have inserted it into this term and the Si28 and S32 terms

**Review by zingale — APPROVED** [review]

**Comment by zingale** [comment]
test_rhs looks good: 
http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2021-02-17-002/test_rhs_C.html

**Pull request merged**